zcash_client_backend/src/wallet.rs

```diff
@@ -296,27 +299,54 @@ impl<NoteRef> ReceivedNote<NoteRef> {
     pub fn output_index(&self) -> u16 {
         self.output_index
     }
-    pub fn note(&self) -> &WalletNote {
+    pub fn note(&self) -> &NoteT {
         &self.note
     }
-    pub fn value(&self) -> NonNegativeAmount {
-        self.note.value()
-    }
     pub fn spending_key_scope(&self) -> Scope {
         self.spending_key_scope
     }
     pub fn note_commitment_tree_position(&self) -> Position {
         self.note_commitment_tree_position
     }
+
+    pub fn traverse_opt<B>(
```

Comment:
So, what I really, really wish for is `traverse`: https://hackage.haskell.org/package/base-4.19.0.0/docs/Prelude.html#v:traverse

```
traverse :: Applicative f => (a -> f b) -> t a -> f (t b)
```

This implementation is `traverse_opt` because it specializes `f` to `Option` (and `t` in this case is the functor `ReceivedNote<NoteRef, _>`).

Reply:
(Option satisfies the `Applicative` constraint)

Reply:
I guess that the more Rust-y way would be to implement `map` and `transpose`?